src/gaxios.ts

```diff
@@ -230,6 +252,49 @@ export class Gaxios {
     return true;
   }
 
+  /**
+   * Applies the interceptors. The request interceptors are applied after the
+   * call to prepareRequest is completed. The response interceptors are applied after the call
+   * to translateResponse.
+   *
+   * @param {T} optionsOrResponse The current set of options or the translated response.
+   *
+   * @returns {Promise<T>} Promise that resolves to the set of options or response after interceptors are applied.
+   */
+  async #applyInterceptors<
```

Comment:
Rather than adding `GaxiosInterceptorType` perhaps we can have separate methods for `#applyInterceptors`; `#applyRequestInterceptors` and `#applyResponseInterceptors` - the logic in `#applyInterceptors` can be cleanly separated by its `if/else` condition

Reply:
That was what I was originally trying to avoid but now that I think about it, makes more sense, single responsibility and all that. Will adjust.